webserver/webserver/src/main/java/io/helidon/webserver/WebServerHelper.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2018 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.webserver;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A set of helper methods for accessing WebServer functionality. This
+ * is primarily used by tests in other packages that need to construct
+ * WebServer objects, but should not know the implementation details
+ * of webserver
+ */
+public class WebServerHelper {
+
+    private WebServerHelper() {
+        //not called
+    }
+
+    /**
+     * Create an instance of RequestHeaders.
+     *
+     * @param headerMap a Map of request headers
+     *
+     * @return a new instance built with the passed headers
+     */
+    public static RequestHeaders constructRequestHeaders(Map<String, List<String>> headerMap) {
+        return new HashRequestHeaders(headerMap);
+    }
```

Comment:
Adding a new class to public API of Helidon seems like a bit of an overkill for a single method. Also I do not like the idea of test support being part of core packages.
Also you can use mocking for testing.

Reply:
I understand not wanting to add this to the public API and I'm open to other suggestions. But I don't like mocking RequestHeaders because the fix is sensitive to WebServer's behavior of RequestHeaders. And I'd be testing the mock more than the fix. 

So maybe this is not well suited for a unit test.

Reply:
Method `requestsJsonData` adds very little logic on top of `bestAccepted`. Wouldn't it be sufficient to test the use case in the `webserver` module and simply this?

Reply:
And I had a bug in that "very little logic" when I first wrote the method. I really want to test the code path for determining if we generate Prometheus or JSON format. At this point I'm thinking this is best left to a functional or integration type test instead of a unit test. I'll remove this unit test.